src/EnergyPlus/PluginManager.hh

```diff
@@ -99,43 +95,65 @@ namespace PluginManagement {
-        bool bHasBeginZoneTimestepAfterInitHeatBalance = false;
         const char * sHookBeginTimestepBeforePredictor = "on_begin_timestep_before_predictor";
-        bool bHasBeginTimestepBeforePredictor = false;
         const char * sHookAfterPredictorBeforeHVACManagers = "on_after_predictor_before_hvac_managers";
-        bool bHasAfterPredictorBeforeHVACManagers = false;
         const char * sHookAfterPredictorAfterHVACManagers = "on_after_predictor_after_hvac_managers";
-        bool bHasAfterPredictorAfterHVACManagers = false;
         const char * sHookInsideHVACSystemIterationLoop = "on_inside_hvac_system_iteration_loop";
-        bool bHasInsideHVACSystemIterationLoop = false;
         const char * sHookEndOfZoneTimestepBeforeZoneReporting = "on_end_of_zone_timestep_before_zone_reporting";
-        bool bHasEndOfZoneTimestepBeforeZoneReporting = false;
         const char * sHookEndOfZoneTimestepAfterZoneReporting = "on_end_of_zone_timestep_after_zone_reporting";
-        bool bHasEndOfZoneTimestepAfterZoneReporting = false;
         const char * sHookEndOfSystemTimestepBeforeHVACReporting = "on_end_of_system_timestep_before_hvac_reporting";
-        bool bHasEndOfSystemTimestepBeforeHVACReporting = false;
         const char * sHookEndOfSystemTimestepAfterHVACReporting = "on_end_of_system_timestep_after_hvac_reporting";
-        bool bHasEndOfSystemTimestepAfterHVACReporting = false;
         const char * sHookEndOfZoneSizing = "on_end_of_zone_sizing";
-        bool bHasEndOfZoneSizing = false;
         const char * sHookEndOfSystemSizing = "on_end_of_system_sizing";
-        bool bHasEndOfSystemSizing = false;
         const char * sHookAfterComponentInputReadIn = "on_end_of_component_input_read_in";
-        bool bHasAfterComponentInputReadIn = false;
         const char * sHookUserDefinedComponentModel = "on_user_defined_component_model";
-        bool bHasUserDefinedComponentModel = false;
         const char * sHookUnitarySystemSizing = "on_unitary_system_sizing";
+        bool bHasBeginNewEnvironment = false;
+        bool bHasAfterNewEnvironmentWarmUpIsComplete = false;
+        bool bHasBeginZoneTimestepBeforeInitHeatBalance = false;
+        bool bHasBeginZoneTimestepAfterInitHeatBalance = false;
+        bool bHasBeginTimestepBeforePredictor = false;
+        bool bHasAfterPredictorBeforeHVACManagers = false;
+        bool bHasAfterPredictorAfterHVACManagers = false;
+        bool bHasInsideHVACSystemIterationLoop = false;
+        bool bHasEndOfZoneTimestepBeforeZoneReporting = false;
+        bool bHasEndOfZoneTimestepAfterZoneReporting = false;
+        bool bHasEndOfSystemTimestepBeforeHVACReporting = false;
+        bool bHasEndOfSystemTimestepAfterHVACReporting = false;
+        bool bHasEndOfZoneSizing = false;
+        bool bHasEndOfSystemSizing = false;
+        bool bHasAfterComponentInputReadIn = false;
+        bool bHasUserDefinedComponentModel = false;
```

Comment:
I think I could actually get rid of these booleans now, and just use the PyObject variables to check for not `nullptr`, or _truthy._